docs/content/guides/dagster/recommended_project_structure.mdx

```diff
@@ -0,0 +1,201 @@
+
+```shell
+├── project_fully_featured
+    ...
+│   ├── assets
+│   │   ├── __init__.py
+│   │   ├── activity_analytics
+│   │   │   ├── __init__.py
+│   │   │   └── activity_forecast.py
+│   │   ├── core
+│   │   │   ├── __init__.py
+│   │   │   ├── id_range_for_time.py
+│   │   │   └── items.py
+│   │   └── recommender
+│   │       ├── __init__.py
+│   │       ├── comment_stories.py
+│   │       ├── recommender_model.py
+│   │       ├── user_story_matrix.py
+│   │       └── user_top_recommended_stories.py
+        ....
+```
+
+---
+
+## For schedules and sensors
+
+In this example, we put sensors and schedules a folder [`sensors`](https://github.com/dagster-io/dagster/blob/master/examples/project_fully_featured/project_fully_featured/sensors). When we build sensors, they are considered as policies for when to trigger a particular job. So, keeping all the policies together helps us understand what can later be used when created the jobs.py file.
+
+- Note: Certain sensors like [run status sensors](/concepts/partitions-schedules-sensors/sensors#run-status-sensors) can listen to multiple jobs and do not trigger a job. We recommend keeping these sensors in the definition as they are often for alerting and monitoring at the code location level.
+
+---
+
+## For resources
+
+Make resources reusable and share them across jobs or asset groups.
+
+In this example, resources (e.g., database connections, Spark sessions, API clients, and I/O managers) are grouped in the [`resources`](https://github.com/dagster-io/dagster/blob/master/examples/project_fully_featured/project_fully_featured/resources/) folder, where they are bound to configuration sets which vary based on environment.
+
+In complex projects, we find it helpful to make resources reusable and configured with pre-defined values via <PyObject object="configured" />. This allows your teammates to use a pre-defined resource set or make changes to shared resources, thus enabling more efficient project development.
+
+This pattern also helps you to easily execute jobs in different environments without code changes. In this example, we dynamically define a code location based on the deployment in [`__init__.py`](https://github.com/dagster-io/dagster/blob/master/examples/project_fully_featured/project_fully_featured/\__init\_\_.py) and are able to keep all code the same across testing, local development, staging, and production. Read more about our recommendations in [Transitioning Data Pipelines from Development to Production](/guides/dagster/transitioning-data-pipelines-from-development-to-production)
+
+---
+
+## For jobs
+
+When using assets based data pipelines, we recommend having a jobs.py file which imports the assets, partitions, sensors, etc. to build each of the jobs.
+
+### When using ops and graphs
+
+This project does not include ops or graphs, if it did, this would be the recommendation on how to structure it.
```

Comment:
```suggestion
This project does not include ops or graphs; if it did, this would be the structure recommendation.
```

🔧 Using commas made this sound like a run-on. Switching to a semi-colon fixes that